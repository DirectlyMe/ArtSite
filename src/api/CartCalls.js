import { serverConfig as config } from "../config/serverConfig";

export async function addToCart(id, quantity = 1, type = "") {
  const data = {
    product_id: id,
    quantity: quantity,
    type: type
  };

  try {
    const response = await fetch(
      `http://${config.IP}:${config.PORT}/cart/add-item`,
      {
        credentials: "include",
        method: "POST",
        mode: "cors",
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json; charset=utf-8"
        }
      }
    );
    console.log(response);
    return true;
  } catch (err) {
    console.log(err);
    return false;
  }
}

export async function getCart() {
  try {
    const response = await fetch(
      `http://${config.IP}:${config.PORT}/cart/get-cart`,
      {
        credentials: "include",
        method: "GET",
        mode: "cors"
      }
    );
    return await response.json();
  } catch (err) {
    return err;
  }
}

export async function removeFromCart(id, type) {
  const data = {
    product_id: id,
    type: type
  };

  try {
    const response = await fetch(
      `http://${config.IP}:${config.PORT}/cart/remove-item`,
      {
        credentials: "include",
        method: "DELETE",
        mode: "cors",
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json; charset=utf-8"
        }
      }
    );
    return await response.json();
  } catch (err) {
    return err;
  }
}
