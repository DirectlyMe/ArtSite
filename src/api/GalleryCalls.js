import { serverConfig as config } from "../config/serverConfig";


export async function getGalleryItems() {
  try {
    const response = await fetch(`${config.IP}:${config.PORT}/gallery/get-gallery`, {
      method: "GET",
      mode: "cors",
      credentials: "same-origin"
    });
    return await response.json();
  }
  catch (err) {
    return err;
  }
}

export async function getSingleItem(itemId) {
  try {
    const response = await fetch(`${config.IP}:${config.PORT}/gallery/getGalleryItem/${itemId}`);
    return await response.json();
  }
  catch (err) {
    return err;
  }
}
