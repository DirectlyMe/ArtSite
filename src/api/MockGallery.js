let roses = require('../Images/Roses.jpg')
let jump = require('../Images/Jump.jpg')
let abstract = require('../Images/abstract.jpg')

export function getGalleryItems() {
    return [
        {
            id: '12345',
            imagePath: roses,
            title: 'Beautiful Roses',
            description: 'test description'
        },
        {
            id: '23451',
            imagePath: jump,
            title: 'Lorem Ipsum',
            description: 'Lorem Ipsum Opsem'
        },
        {
            id: '23421',
            imagePath: abstract,
            title: 'Lorem Ipsum',
            description: 'Lorem Ipsum Opsem'
        },
    ]
}
