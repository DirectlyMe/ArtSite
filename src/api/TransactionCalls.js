 import { serverConfig as config } from "../config/serverConfig";

export async function postTransactionInfo(userInfo) {
    try {
        const response = fetch(`${config.IP}:${config.PORT}/transaction/post-transaction`, {
            credentials: "include",
            method: "POST",
            mode: "cors",
            body: JSON.stringify(userInfo),
            headers: {
                "Content-Type": "application/json; charset=utf-8"
            }
        });

        return response;
    } catch (err) {
        console.log(err);
        return false;
    }
}
