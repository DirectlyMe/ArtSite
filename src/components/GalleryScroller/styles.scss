.scroller {
  display: flex;
  flex-direction: column;
  width: 100%;
  overflow-y : scroll;
  overflow-x: hidden; 
  padding-top: 30px;
  padding-right: 17px;
  margin-left: 3%;
  text-align: center;

  ::-webkit-scrollbar {
    display: none;
  }
}