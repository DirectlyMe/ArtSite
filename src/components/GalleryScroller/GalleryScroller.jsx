import React, { Component, Suspense, lazy } from "react";
import GalleryContext from "../../GalleryContext";
import "./styles.scss";

const ScrollerItem = lazy(() => import("../ScollerItem"));

class GalleryScroller extends Component {
  render() {
    const galleryItems = this.context.galleryItems.map(item => (
      <Suspense key={item.title} fallback={<div>loading...</div>}>
        <ScrollerItem item={item} />
      </Suspense>
    ));
    
    return (
      <div style={{ overflow: "hidden" /* needed to hide scrollbar */ }}>
        <div className="scroller" style={{ height: window.innerHeight - 120 }}>
          {galleryItems}
        </div>
      </div>
    );
  }
}
GalleryScroller.contextType = GalleryContext;

export default GalleryScroller;
