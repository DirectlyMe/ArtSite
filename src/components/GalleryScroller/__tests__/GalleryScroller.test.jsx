import React from "react";
import { MemoryRouter } from "react-router-dom";
import GalleryScroller from "../GalleryScroller";
import renderer from "react-test-renderer";
import { getGalleryItems } from "../../../api/GalleryCalls";

describe("The GalleryScroller", () => {
    it("Renders as expected", async () => {
        const galleryItems = await getGalleryItems();
        const tree = renderer
            .create(
                <MemoryRouter>
                    <GalleryScroller
                        galleryItems={galleryItems}
                        height={window.innerHeight}
                    />
                </MemoryRouter>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
