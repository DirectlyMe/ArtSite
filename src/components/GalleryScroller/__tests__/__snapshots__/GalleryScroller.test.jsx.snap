// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GalleryScroller Renders as expected 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <div
    className="scroller"
    style={
      Object {
        "height": 683,
      }
    }
  />
</div>
`;
